Fix NREs in DiscordService.

RoosterDragon opened the pull request:
Handle the client being null. Previously, a service could be created with a null client. This would leads to NREs when invoking the static Update methods. Now we provide a static helper to get the instance created for the mod. This avoids two instances of the DiscordService being created (a static one, and an instance created in the mod manifest).

Fixes #21275

PunkPun reviewed (commented):
When I launch a dedicated server discord detects it. While on bleed it doesn't

RoosterDragon commented:
> When I launch a dedicated server discord detects it. While on bleed it doesn't

You are correct. There's two instances going on here. One created by the mod manifest and the static one created if you use the static helper methods. A bit ugly.

Streamlined to use one instance, and ensure that instance is able to deal with a null client.

anvilvapre commented:
There was a reason to allow for disabling the service. #19343
To me it seems like an implementation issue in the client library. 

Many players on slower system would benefit from it being disabled.

RoosterDragon commented:
The disabling logic `if (!Game.Settings.Game.EnableDiscordService) return;` is still there in the constructor.

PunkPun commented on OpenRA.Mods.Common/DiscordService.cs:
```diff
@@ -206,29 +190,26 @@ void UpdateParty(int players, int slots)
-			Service?.client.UpdateDetails(details);
+			var manifest = modData.Manifest;
+			if (manifest.Contains<DiscordService>())
+				return manifest.Get<DiscordService>();
```
why can't we cache he instance like the previous script did?

RoosterDragon replied:
On thinking about it, we can continue to do that. Adjusted back to that pattern.

RoosterDragon committed:
Fix NREs in DiscordService.

Handle the client being null. Previously, a service could be created with a null client. This would leads to NREs when invoking the static Update methods. Now we guard against a null client.

PunkPun reviewed (approved):
LGTM

PunkPun commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/088ee336dccbf6f9cd049959bf049da9601921a0...61ede439d8f4ccb10288e683252392de95c3cd68)
